bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/Utils.java

```diff
@@ -455,4 +461,42 @@ public static Object getFiniteBalValue(TomlNode tomlNode, Set<TomlNode> visitedN
         }
 
     }
+
+    static boolean isEquivalentTomlTypeFromTuple(TomlType kind, Type type, String variableName) {
```

Comment:
Shall we move this to getEffectiveTomlTypemethod it self so inline tables will work for maps and table as well.